### ISSUE TITLE
Menu settings of custom layouts not cloned properly

### ISSUE THREAD

**Issue by Al-Rozhkov** [backdrop/backdrop-issues#2673]
Steps to reproduce:

1. Create custom layout "News" with path `news`.
2. Choose "Normal menu entry" option in menu settings of newly added layout.
3. Clone "News" layout with name "Events", machine name `events` and path `events`.

![menu-settings-of-cloned-layouts](https://cloud.githubusercontent.com/assets/16973264/25772146/365632ec-3275-11e7-8287-de6d0a81d182.jpg)

Menu settings button of original layout is disappeared. And menu settings button of cloned layout leads to settings of the original.

Also there isn't config json file for `events` menu settings, only `layout.menu_item.news.json` is there.

---
~PR by @Al-Rozhkov: https://github.com/backdrop/backdrop/pull/1872~
PR by @Al-Rozhkov and @quicksketch: https://github.com/backdrop/backdrop/pull/2574 (with tests)

**Comment by Al-Rozhkov** [comment]
PR works for me, but definitely needs review of experienced developers.

**Comment by quicksketch** [comment]
Nice find @Al-Rozhkov. We need to add a test for this cloning functionality. We already test cloning layouts of existing paths (e.g. `node/%`) but we have not tested cloning layout-owned paths. PR fix looks good.

**Comment by quicksketch** [comment]
I'm marking this needs work as we need to add tests. The fix in the PR is great.

**Comment by Al-Rozhkov** [comment]
I can't write tests yet, so unassigning myself.

**Comment by jenlampton** [comment]
I just ran into this bug today, it's pretty major as both my old layout and my new one are borked after cloning. I wonder... should a fix for something this major be held up on tests? This has been very broken in backdrop for the last 7 releases, when we've had a perfectly good fix lying around.

**Comment by jenlampton** [comment]
@quicksketch any chance we can commit a fix without tests? This is seriously broken atm, and we have a fix sitting right here :) it pains me to re-apply this patch with every release.

**Comment by jenlampton** [comment]
bump :)

**Comment by klonos** [comment]
I am adding the "help wanted" tag for anyone that can speak tests. Pity to hold this back so long, but I understand.

**Comment by laryn** [comment]
Another RTBC -- PR worked for me.

**Comment by quicksketch** [comment]
I made a new PR at https://github.com/backdrop/backdrop/pull/2574 that includes tests.

**Comment by herbdool** [comment]
Code looks good.

**Comment by quicksketch** [comment]
Thanks @herbdool for the review! Thanks @Al-Rozhkov, @laryn, @jenlampton, and @klonos for the initial patch and testing. I've merged https://github.com/backdrop/backdrop/pull/2574 into 1.x and 1.12.x.
